service/auth/auth.go

```diff
@@ -279,23 +293,80 @@
 		return nil, status.Errorf(codes.Unauthenticated, "login failed")
 	}
 
-	var token string
+	if token, err = issueToken(s.apiKey, user.Username, user.FullName, user.Email, expiry); err != nil {
+		return nil, status.Errorf(codes.Internal, "token issue failed: %v", err)
+	}
+
+	if refreshToken, err = issueRefreshToken(s.apiKey, user.Username); err != nil {
+		return nil, status.Errorf(codes.Internal, "refresh token issue failed: %v", err)
+	}
 
-	var expiry = time.Now().Add(time.Hour * 24)
+	response = &auth.TokenResponse{
+		AccessToken:  token,
+		RefreshToken: refreshToken,
+		TokenType:    "Bearer",
+		Expiry:       timestamppb.New(expiry),
+	}
 
-	if token, err = s.issueToken(u.Username, u.FullName, u.Email, expiry); err != nil {
-		return nil, status.Errorf(codes.Internal, "token issue failed: %v", err)
+	return response, nil
+}
+
+// Token aims to be a OAuth 2.0 compliant token endpoint. Currently, only the refresh_token grant type
+// is supported. This function only takes care of the application behavior, the actual implementation of
+// a HTTP endpoint around this functionality is provided by the gRPC gateway and is modified by the
+// OAuthErrorHandler function, which needs to be registed using runtime.WithErrorHandler in the gRPC
+// gateway.
+func (s *Service) Token(_ context.Context, req *auth.TokenRequest) (response *auth.TokenResponse, err error) {
+	var (
+		token  string
+		user   *auth.User
+		claims jwt.RegisteredClaims
+		expiry = time.Now().Add(DefaultPeriodOfValidity)
+	)
+
+	if req == nil {
```

Comment:
Should we really check a gRPC request for being nil? Can it even be nil? I am pretty sure we didn't do it before/everywhere